addons/mail/static/src/models/attachment_link_preview/attachment_link_preview.js

```diff
@@ -0,0 +1,140 @@
+
+    class AttachmentlinkPreview extends dependencies['mail.model'] {
+
+        /**
+         * @override
+         */
+        _created() {
+            // Bind necessary until OWL supports arrow function in handlers: https://github.com/odoo/owl/issues/876
+            this.onClickUnlink = this.onClickUnlink.bind(this);
+            this.onDeleteConfirmDialogClosed = this.onDeleteConfirmDialogClosed.bind(this);
+        }
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Handles the click on delete attachment and open the confirm dialog.
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickUnlink(ev) {
+            ev.stopPropagation();
+            if (this.attachment) {
+                if (this.attachment.isLinkedToComposer) {
+                    this.component.trigger('o-attachment-removed', { attachmentLocalId: this.attachment.localId });
+                    this.attachment.remove();
+                } else {
+                    this.update({ hasDeleteConfirmDialog: true });
+                }
+            }
+        }
+
+        /**
+         * Synchronize the `hasDeleteConfirmDialog` when the dialog is closed.
+         */
+        onDeleteConfirmDialogClosed() {
+            this.update({
+                hasDeleteConfirmDialog: false,
+            });
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {number}
+         */
+        _computeHeight() {
+            return this.width;
+        }
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeImageUrl() {
+            if (!this.attachment) {
+                return;
+            }
+            if (!this.attachment.accessToken && this.attachment.originThread && this.attachment.originThread.model === 'mail.channel') {
+                return `/mail/channel/${this.attachment.originThread.id}/image/${this.attachment.id}/${this.width}x${this.height}`;
+            }
+            const accessToken = this.attachment.accessToken ? `?access_token=${this.attachment.accessToken}` : '';
+            return `/web/image/${this.attachment.id}/${this.width}x${this.height}${accessToken}`;
+        }
+
+        /**
+         * @private
+         * @returns {number}
+         */
+        _computeWidth() {
+            if (!this.attachmentList) {
+                return;
+            }
+            if (this.attachmentList.composer) {
+                return 50;
+            }
+            if (this.attachmentList.thread) {
+                return 160;
+            }
+            if (this.attachmentList.message) {
+                return 300;
+            }
```

Comment:
I didn't test yet, but we probably do not want link preview to be as big as normal image.